candle-core/src/metal_backend/device.rs

```diff
@@ -99,48 +132,85 @@ impl MetalDevice {
         &self.command_queue
     }
 
+    pub fn seed(&self) -> Arc<Mutex<Buffer>> {
+        self.seed.clone()
+    }
+
+    /// Returns the current active command buffer, if the buffer is not present, this will allocate one
     pub fn command_buffer(&self) -> Result<CommandBuffer> {
         let mut command_buffer_lock = self.command_buffer.try_write().map_err(MetalError::from)?;
-        let mut command_buffer = command_buffer_lock.to_owned();
-        let mut index = self
-            .command_buffer_index
-            .try_write()
-            .map_err(MetalError::from)?;
-        if *index > self.compute_per_buffer {
-            command_buffer.commit();
-            command_buffer = self.command_queue.new_command_buffer().to_owned();
-            *command_buffer_lock = command_buffer.clone();
-            *index = 0;
+        let command_buffer = command_buffer_lock.to_owned();
+
+        if let Some(command_buffer) = command_buffer {
+            Ok(command_buffer)
+        } else {
+            let new_command_buffer = self.command_queue.new_command_buffer().to_owned();
+            *command_buffer_lock = Some(new_command_buffer);
+            Ok(command_buffer_lock.to_owned().unwrap())
```

Comment:
This unwrap seems a bit weird, why not cloning `new_command_buffer` before storing it in `command_buffer_lock` and returning this cloned value instead?